lib/mix/test/mix/tasks/escript_test.exs

```diff
@@ -94,6 +106,51 @@ defmodule Mix.Tasks.EscriptTest do
       Mix.Tasks.Escript.Build.run []
       assert_received {:mix_shell, :info, ["Generated escript escriptest with MIX_ENV=dev"]}
       assert System.cmd("escript", ["escriptest"]) == {"FROM CONFIG\n", 0}
+      assert count_abstract_code("escriptest") == 0
+    end
+  end
+
+  test "generate escript with debug information" do
+    Mix.Project.push EscriptWithDebugInfo
+
+    in_fixture "escripttest", fn ->
+      Mix.Tasks.Escript.Build.run []
+      assert_received {:mix_shell, :info, ["Generated escript escripttestwithdebuginfo with MIX_ENV=dev"]}
+      assert System.cmd("escript", ["escripttestwithdebuginfo"]) == {"TEST\n", 0}
+      assert count_abstract_code("escripttestwithdebuginfo") > 0
+
+      Mix.Tasks.Escript.Build.run []
+      refute_received {:mix_shell, :info, ["Generated escript escripttestwithdebuginfo with MIX_ENV=dev"]}
+    end
+  end
+
+  defp count_abstract_code(escript_filename) do
+    escript_filename
+    |> get_beams()
+    |> Enum.map(fn {_, beam} -> get_abstract_code(beam) end)
```

Comment:
The solution I sent should have worked. Instead of:

    |> Enum.map(fn {_, beam} -> get_abstract_code(beam) end)
    |> Enum.reject(&is_nil/1)
    |> Enum.count

You could do:

    |> Enum.count(fn {_, beam} -> get_abstract_code(beam) end)

As:

```iex
iex> Enum.count [:a, nil, :b], fn x -> x end
2
```